File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(string bin1, string bin2) 
{
    int d1 = 0;
    int d2 = 0;

    for(int i=0; i<bin1.size(); i++)
    {
        if(bin1[i] == '1')
        {
            d1 += (1 << bin1.size() - 1 - i);
        }
    }
    
    for(int i=0; i<bin2.size(); i++)
    {
        if(bin2[i] == '1')
        {
            d2 += (1 << bin2.size() - i - 1);
        }
    }
    
    string result = "";
    int sum = d1 + d2;
    cout << sum;
    if(sum == 0)
    {
        result = "0";
    }
    else
    {    
        while(sum > 0)
        {
            result += (sum%2 == 0) ? '0' : '1';
            sum /= 2;
        }
    }
    reverse(result.begin(), result.end());
    return result;
}

File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.cs
using System;

public class Solution 
{
    public string solution(string bin1, string bin2) 
    {
        int bin1_int = 0;
        int bin2_int = 0;
        for(int i=0; i<bin1.Length; i++) {
            switch(bin1[i].ToString()) {
                case "1":
                    bin1_int += (int)Math.Pow(2,bin1.Length - i - 1); 
                    break;
                case "0":
                    break;
            }
        }
        for(int i=0; i<bin2.Length; i++) {
            switch(bin2[i].ToString()) {
                case "1":
                    bin2_int += (int)Math.Pow(2,bin2.Length - i - 1); 
                    break;
                case "0":
                    break;
            }
        }
        int bin_int = bin1_int + bin2_int;
        
        return Convert.ToString(bin_int, 2);;
    }
}